apps/server/src/api-data/automation/automation.service.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  isHTTPOutput,
+  isOSCOutput,
+  type AutomationFilter,
+  type AutomationOutput,
+  type FilterRule,
+  type RuntimeStore,
+  type TimerLifeCycle,
+} from 'ontime-types';
+
+import { emitOSC } from './clients/osc.client.js';
+import { emitHTTP } from './clients/http.client.js';
+import { getAutomations, getBlueprints } from './automation.dao.js';
+import { getState, type RuntimeState } from '../../stores/runtimeState.js';
+
+/**
+ * Exposes a method for triggering actions based on a TimerLifeCycle event
+ */
+export function triggerAutomations(event: TimerLifeCycle, state: RuntimeState) {
+  const automations = getAutomations();
+  const triggerAutomations = automations.filter((automation) => automation.trigger === event);
+  if (triggerAutomations.length === 0) {
+    return;
+  }
+
+  const blueprints = getBlueprints();
+  if (Object.keys(blueprints).length === 0) {
+    return;
+  }
+
+  triggerAutomations.forEach((automation) => {
+    const blueprint = blueprints[automation.blueprintId];
+    if (!blueprint) {
+      return;
+    }
+    const shouldSend = testConditions(blueprint.filters, blueprint.filterRule, state);
+    if (shouldSend) {
+      send(blueprint.outputs, state);
+    }
+  });
+}
+
+export function testOutput(payload: AutomationOutput) {
+  send([payload]);
+}
+
+/**
+ * Checks whether the automation conditions are met
+ */
+export function testConditions(
+  filters: AutomationFilter[],
+  filterRule: FilterRule,
+  state: Partial<RuntimeStore>,
+): boolean {
+  if (filters.length === 0) {
+    return true;
+  }
+
+  if (filterRule === 'all') {
+    return filters.every((filter) => evaluateCondition(filter));
+  }
+
+  return filters.some((filter) => evaluateCondition(filter));
+
+  function evaluateCondition(filter: AutomationFilter): boolean {
+    const { field, operator, value } = filter;
+    const fieldValue = state[field];
+
+    // TODO: if value is empty string, the user could be meaning to check if the value does not exist
+    switch (operator) {
```

Comment:
@alex-Arc i have been wondering if we could come up with a few examples and fine tune the operators?